File: practicas/src/practica3/controllers/UserController.java
package practica3.controllers;

import practica3.model.User;
import practica3.services.impl.OccasionalBuyer;
import practica3.services.impl.WholesaleBuyer;
import practica3.services.UserService;
import practica3.services.WelcomeUser;
import practica3.services.impl.UserServiceImp;

import javax.swing.*;
import javax.xml.namespace.QName;

public class UserController {

    UserService userService = new UserServiceImp();

    public UserController() {
    }

    public UserController(UserService userService) {
        this.userService = userService;
    }

    //METHODS

    public void addUser() {
        String name;
        String id;
        String address;
        String phoneNumber;
        String email;
        do {
            name = JOptionPane.showInputDialog("Write the name");
        } while (validateField(name));
        do {
            id = JOptionPane.showInputDialog("Write the id");
        } while (validateField(id));
        do {
            address = JOptionPane.showInputDialog("Write the address");
        } while (validateField(address));
        do {
            phoneNumber = JOptionPane.showInputDialog("Write the phone number");
        } while (validateField(phoneNumber));
        do {
            email = JOptionPane.showInputDialog("Write the email");
        } while (validateField(email));
        WelcomeUser userType = showUserTypeMenu();
        userService.addUser(name, id, address, phoneNumber, email, userType);
    }

    public void updateUser() {
        User user;
        do {
            user = findUser();
        } while (user == null);
        String name = (String)JOptionPane.showInputDialog(null, "Please enter the new name","Update "+user.getName()
                ,JOptionPane.QUESTION_MESSAGE,null,null,user.getName());
        String id = (String)JOptionPane.showInputDialog(null, "Please enter the new id","Update "+user.getName()
                , JOptionPane.QUESTION_MESSAGE,null,null,user.getId());
        String address = (String)JOptionPane.showInputDialog(null, "Please enter the new address","Update "+user.getName()
                , JOptionPane.QUESTION_MESSAGE,null,null,user.getAddress());
        String phoneNumber = (String)JOptionPane.showInputDialog(null, "Please enter the new phoneNumber","Update "+user.getName()
                , JOptionPane.QUESTION_MESSAGE,null,null,user.getPhoneNumber());
        String email = (String)JOptionPane.showInputDialog(null, "Please enter the new email","Update "+user.getName()
                , JOptionPane.QUESTION_MESSAGE,null,null,user.getEmail());
        userService.updateUser(name, id, address, phoneNumber, email, user);
    }

    private User findUser() {
        String id = JOptionPane.showInputDialog("Write the id to search the user");
        return userService.findUser(id);
    }

    private boolean validateField(String data) {
        if(data.isBlank()){
            JOptionPane.showMessageDialog(null, "You have to enter a name","ERROR",JOptionPane.ERROR_MESSAGE );
            return true;
        }else {
            return false;
        }
    }

    public void deleteUser(){
        String id = JOptionPane.showInputDialog("Write the id to search the user you want to delete");
        User user = userService.findUser(id);
        userService.deleteUser(user);
    }

    private WelcomeUser showUserTypeMenu () {
            WelcomeUser type = null;
            String[] options = new String[]{"Occasional User", "Wholesale Buyer"};
            int response = JOptionPane.showOptionDialog(null, "Please select an option", "Te Lo Consigo Store",
                    JOptionPane.PLAIN_MESSAGE, JOptionPane.PLAIN_MESSAGE,
                    null, options, options[0]);

            switch (response) {
                case 0:
                    return new OccasionalBuyer();
                case 1:
                    return new WholesaleBuyer();
            }
            return null;
    }
}

File: ParcialAnalisis/Controllers/UserController.py
from Services.Imp.User.UserServiceImpl import UserServiceImpl
from Services.Imp.User.WholeSaleBuyer import WholeSaleBuyer
from Services.Imp.User.OccasionalBuyer import OccasionalBuyer
from Services.CrudOperation import CrudOperation

'''
This class manage the data inputs related to the Class: User

This class sends all data inputs to his instance of the class: UserService
'''


class UserController(CrudOperation):

    userService: UserServiceImpl

    def __init__(self):
        self.userService = UserServiceImpl()

    '''
    This method get all the data inputs to create an user and send it to userService
    '''
    def create(self):
        name = input("write the name of the new User")
        id = input("Write the id of the user")
        address = input("Write the address of the user")
        phoneNumber = input("Write the phone number of the user")
        email = input("Write the email od the user")
        userType = self.askUserType()
        self.userService.addUser(name, id, address, phoneNumber, email, userType)

    '''
    This method get all the data inputs to update an user and send it to userService
    '''
    def update(self):
        user = self.get()
        name = input("write the new name. Actual: " + user.name)
        id = input("Write the new id of the user. Actual: " + user.id)
        address = input("Write the new address of the user. Actual: " + user.address)
        phoneNumber = input("Write the new phone number of the user. Actual: " + user.phoneNumber)
        email = input("Write the new email of the user. Actual: " + user.email)
        userType = self.askUserType()
        self.userService.updateUser(name, id, address, phoneNumber, email, userType, user)

    '''
    this method get an input in order to delete an specified user
    '''
    def detele(self):
        pass

    '''
    this method get an input in order to find and return an specified user
    '''
    def get(self):
        id = input("Write the id of the user you want to update")
        return self.userService.findUser(id)

    '''
        this method get an input in order to delete an specified user
    '''
    def askUserType(self):
        opcion = input("Which user are u?\n1.WholeSale Buyer\n2.Occasional Buyer")
        if opcion == "1":
            return WholeSaleBuyer()
        elif opcion == "2":
            return OccasionalBuyer()
        else:
            print("Please Select a valid option")






